tests/common/mod.rs

```diff
@@ -0,0 +1,228 @@
+use cosmwasm_std::{
+    from_binary,
+    testing::{MockApi, MockQuerier, MockStorage},
+    to_binary, Addr, Api, BalanceResponse, BankMsg, BankQuery, Binary, BlockInfo, CosmosMsg,
+    CustomQuery, Empty, MemoryStorage, Querier, Storage, Timestamp,
+};
+use cw_multi_test::{
+    App, AppBuilder, AppResponse, BankKeeper, BankSudo, CosmosRouter, FailingDistribution,
+    FailingStaking, Module, Router, SudoMsg, WasmKeeper,
+};
+use schemars::JsonSchema;
+use sei_cosmwasm::{Epoch, EpochResponse, SeiMsg, SeiQuery, SeiQueryWrapper};
+use sei_tester::{
+    contract::{execute, instantiate, query},
+    msg::{ExecuteMsg, InstantiateMsg, QueryMsg},
+};
+use serde::de::DeserializeOwned;
+use std::{fmt::Debug, marker::PhantomData};
+
+use anyhow::Result as AnyResult;
+
+pub struct SeiModule(PhantomData<(SeiMsg, SeiQuery)>);
+
+impl SeiModule {
+    pub fn new() -> Self {
+        SeiModule(PhantomData)
+    }
+}
+
+impl Default for SeiModule {
+    fn default() -> Self {
+        Self::new()
+    }
+}
+
+impl Module for SeiModule {
+    type ExecT = SeiMsg;
+    type QueryT = SeiQueryWrapper;
+    type SudoT = Empty;
+
+    fn execute<ExecC, QueryC>(
+        &self,
+        api: &dyn Api,
+        storage: &mut dyn Storage,
+        router: &dyn CosmosRouter<ExecC = ExecC, QueryC = QueryC>,
+        block: &BlockInfo,
+        sender: Addr,
+        msg: Self::ExecT,
+    ) -> AnyResult<AppResponse>
+    where
+        ExecC: Debug + Clone + PartialEq + JsonSchema + DeserializeOwned + 'static,
+        QueryC: CustomQuery + DeserializeOwned + 'static,
+    {
+        match msg {
+            SeiMsg::CreateDenom { subdenom } => {
+                let denom = format!("factory/{}/{}", sender, subdenom);
```

Comment:
can we refactor these message handlers into helpers for cleanliness / maintainability?